sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,260 @@
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.adi_cv import ADICVTransformer
+
+
+class CategoryCompositor(BaseForecaster):
+    """Compositor that utilizes varying forecasters by time series data's nature.
+
+    Applies a series-to-primitives transformer on a given time series and utilizes
+    generated primitive value to apply the most appropriate forecaster to the
+    given series.
+
+    Parameters
+    ----------
+    forecasters : dict[sktime forecasters]
+        dict of forecasters with the key corresponding to categories generated
+        by the given transformer and the value corresponding to a sktime forecaster.
+
+    transformer : sktime transformer, default = ADICVTransformer()
+        A series-to-primitives sk-time transformer that generates a value
+        which can be used to quantify a choice of forecaster for the time series.
+
+    Raises
+    ------
+    AssertionError: If a valid transformer (an instance of BaseTransformer)
+    is not passed or if valid forecasters (instances of BaseForecaster) is not given.
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.Series",
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": True,
+        "capability:pred_int": False,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+        "python_version": None,
+        "python_dependencies": [
+            "BaseForecaster",
+            "BaseTransformer" "ADICVTransformer",
+        ],
+    }
+
+    def __init__(self, forecasters, transformer=ADICVTransformer):
+        # saving arguments to object storage
+        self.transformer = transformer
+        self.forecasters = forecasters
+
+        super().__init__()
+
+        # validating passed arguments
+        assert isinstance(transformer, BaseTransformer)
+
+        for forecaster in forecasters.values():
+            assert isinstance(forecaster, BaseForecaster)
+
+        # All checks OK!
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : Pd.Series
+            The target time series to which we fit the data.
+
+        fh : ForecastingHorizon | None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+
+        X : Pd.Series | None, optional (default=None)
+            No exogenous variables are used for this.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        # reset the state of the forecaster
+        self.reset()
```

Comment:
input checks and logic does not need to be carried out here. This is done in the boilerplate layer, in `BaseForecatser.fit`.

Reply:
Ah, got it. Removing the checks from all relevant places!

Reply:
have you done that? still seems like they are there

Reply:
I was at work at the time, but I've made the changes now